### ISSUE TITLE
3.7.4 layout constraint issues?

### ISSUE THREAD

**Issue by zmorris** [rs/SDWebImage#1431]
Hi not sure precisely what's causing this, but I git bisected and version tested my CocoaPods until I discovered that the 3.7.3 to 3.7.4 pod update caused one of my auto layout views to break.  The height of a UITabBar nested in a UITableViewCell went from 49 to 63, possibly because the UIImageView above it changed size in a different way than before.  This could be due to all sorts of reasons, but my gut feeling is that perhaps SDWebImage set the UIImageView's image with slightly different timing so that a call like layoutIfNeeded() changed the frames and then never resized to accommodate the new image size.  This most likely happened near commits https://github.com/rs/SDWebImage/commit/a45b222e7043bf1aa247aab8d1133a36781e21f3 or https://github.com/rs/SDWebImage/commit/6c650f910ef0624e39986075e12b06260d880d40 because we git update about once a week.  Just thought I'd report this while these commits are fresh in everyone's minds in case something rings a bell.  Thanks for all your efforts, this is a great lib!

**Comment by mythodeia** [comment]
the constraints can be found [here](https://github.com/rs/SDWebImage/blob/master/SDWebImage/UIImageView%2BWebCache.m#L174)

if you revert to 3.7.3 is your issue resolved?
can i ask why do you have a UITabBar inside a uitableviewcell? :P

**Comment by zmorris** [comment]
Hi just wanted to follow up, I solved the issue and it had nothing to do with SDWebImage.  It was caused by content hugging and compression resistance priorities in Interface Builder.  The issue was that the UITabBar had a content hugging priority of 250 but an image had a content hugging priority priority of 251 (or vice versa, I might not be remembering right).

Anyway, once SDWebImage received the image and called setNeedsLayout(), the layout broke because the UIImageView tried to expand around it.  I think that 2.7.4 may have slightly different timing so the layout didn't break at first, but after a while it started happening again.

I hadn't been aware that UIImageView could resize this way based on its content image.  I fixed it by setting content hugging and compression resistance priorities on the UIImageView to 125 in Interface Builder so that the surrounding layout prevents the UIImageView from growing when its image changes.

P.S. The UITabBar was in the cell for legacy reasons, to avoid putting it in the header view or having to refactor the UITableViewController to be a UIViewController with UITableViewDelegate, UITableViewDataSource and nested UITableView (to make room for the UITabBar above the table).
